### PR TITLE
[FLINK-29895][Connectors/DynamoDB] Migrate integration tests to unit tests and improve unit test coverage

### PR DESCRIPTION
**Pull request by hlteoh37**
## What is the purpose of the change
- Migrate some integration tests to unit tests
- Improve unit test coverage.

## Verifying this change
- Build locally and verified unit tests and integ tests pass.
- Code coverage improved from `82% classes, with 64% lines covered` to `88% classes, 88% lines covered`

## Does this pull request potentially affect one of the following parts:
* Dependencies (does it add or upgrade a dependency): no
* The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
* The serializers: no
* The runtime per-record code paths (performance sensitive): no
* Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
* The S3 file system connector: no

## Documentation
* Does this pull request introduce a new feature? no
* If yes, how is the feature documented? n/a

### CODE REVIEW

File: flink-connector-dynamodb/src/test/java/org/apache/flink/connector/dynamodb/sink/DynamoDbSinkTest.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
+package org.apache.flink.connector.dynamodb.sink;
```
Comment (dannycranmer) [inline flink-connector-dynamodb/src/test/java/org/apache/flink/connector/dynamodb/sink/DynamoDbSinkTest.java:1]: Missing copyright

File: flink-connector-dynamodb/src/test/java/org/apache/flink/connector/dynamodb/sink/DynamoDbSinkTest.java [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+import org.apache.flink.connector.base.sink.writer.TestSinkInitContext;
+
+import org.junit.Test;
```
Comment (dannycranmer) [inline flink-connector-dynamodb/src/test/java/org/apache/flink/connector/dynamodb/sink/DynamoDbSinkTest.java:5]: Please use Junit 5

**Review by dannycranmer — CHANGES_REQUESTED** [review]

File: flink-connector-dynamodb/src/test/java/org/apache/flink/connector/dynamodb/sink/DynamoDbSinkTest.java (hunk 1 above)
Comment (hlteoh37) [reply flink-connector-dynamodb/src/test/java/org/apache/flink/connector/dynamodb/sink/DynamoDbSinkTest.java:19]: resolved

File: flink-connector-dynamodb/src/test/java/org/apache/flink/connector/dynamodb/sink/DynamoDbSinkTest.java (hunk 2 above)
Comment (hlteoh37) [reply flink-connector-dynamodb/src/test/java/org/apache/flink/connector/dynamodb/sink/DynamoDbSinkTest.java]: done. good spot.

**Review by dannycranmer — APPROVED** [review]

**Pull request merged**